communication/nats/connection_wrap.go

```diff
@@ -72,40 +69,35 @@ func NewConnection(serverURIs ...string) (*ConnectionWrap, error) {
 	return connection, nil
 }
 
-// OpenConnection creates connection instances and connects instantly
-func OpenConnection(serverURIs ...string) (*ConnectionWrap, error) {
-	connection, err := NewConnection(serverURIs...)
-	if err != nil {
-		return connection, err
-	}
-
-	return connection, connection.Open()
-}
-
 // ConnectionWrap defines wrapped connection to NATS server(s)
 type ConnectionWrap struct {
 	*nats_lib.Conn
 	servers     []string
 	removeRules func()
+	onClose     func()
 }
 
-// Open starts the connection
-func (c *ConnectionWrap) Open() (err error) {
+func (c *ConnectionWrap) connectOptions() nats_lib.Options {
 	options := nats_lib.GetDefaultOptions()
 	options.Servers = c.servers
-	options.MaxReconnect = BrokerMaxReconnect
-	options.ReconnectWait = BrokerReconnectWait
-	options.Timeout = BrokerTimeout
+	options.MaxReconnect = -1
+	options.ReconnectWait = 1 * time.Second
+	options.Timeout = 5 * time.Second
 	options.PingInterval = 10 * time.Second
-	options.DisconnectedCB = func(nc *nats_lib.Conn) { log.Warn().Msg("Disconnected") }
-	options.ReconnectedCB = func(nc *nats_lib.Conn) { log.Warn().Msg("Reconnected") }
+	options.ClosedCB = func(conn *nats_lib.Conn) { log.Warn().Msg("NATS: connection closed") }
+	options.DisconnectedCB = func(nc *nats_lib.Conn) { log.Warn().Msg("NATS: disconnected") }
+	options.ReconnectedCB = func(nc *nats_lib.Conn) { log.Warn().Msg("NATS: reconnected") }
+	return options
+}
 
+// Open starts the connection
+func (c *ConnectionWrap) Open() (err error) {
```

Comment:
After connection factory `BrokerConnector` was introduced - is it a point to have `ConnectionWrap.Open/Close()` or `ConnectionWrap` at all? Because it had purposes - to encapsulate connection open and reconnection logic, which now is done by connector

Reply:
Agreed. I left it deprecated for test compatibility. We should rewrite those unit tests using a normal broker instead of mock festival.